### PR TITLE
[types] Rename Transaction::WriteSet to WaypointWriteSet.

### PR DESCRIPTION
**Pull request by runtian-zhou**
## Motivation

Rename  `Transaction::WriteSet` to `WaypointWriteSet`. In the future, `WaypointWriteSet` will be the entry point for applying writesets manually after waypoint. Genesis transaction will be the very first example of this mechanism.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### CODE REVIEW

File: json-rpc/src/views.rs [hunk 1]
```diff
@@ -221,7 +221,7 @@ impl From<Transaction> for TransactionDataView {
                     timestamp_usecs: x.1,
                 })
             }
-            Transaction::WriteSet(_) => Ok(TransactionDataView::WriteSet {}),
+            Transaction::AuthenticatedWriteSet(_) => Ok(TransactionDataView::WriteSet {}),
```
Comment (christinacelee) [inline json-rpc/src/views.rs:224]: @phoenix-antigravity @thefallentree should we rename `TransactionDataView::WriteSet` to `TransactionDataView::AuthenticatedWriteSet` as well?

File: language/tools/vm-genesis/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        stdlib_modules,
+        VMPublishingOption::Open,
+    ))
+    .expect("Generating genesis block failed")
+}
+
+fn main() {
+    println!("Creating genesis binary blob at {}", GENESIS_LOCATION,);
+
+    let mut file = File::create(GENESIS_LOCATION).unwrap();
```
Comment (christinacelee) [inline language/tools/vm-genesis/src/main.rs:35]: nit: maybe expect instead of unwrap?

File: language/tools/vm-genesis/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,55 @@
+// Copyright (c) The Libra Core Contributors
```
Comment (sblackshear) [inline language/tools/vm-genesis/src/main.rs:1]: This is deleted in master; please don't bring it back!

Comment (davidiw) [reply language/tools/vm-genesis/src/main.rs:1]: You have done great things, Sam, great things!

**Comment by matbd** [comment]
Yay, my new test is working!

```
--- test_recorded_formats_did_not_change stdout ----
thread 'test_recorded_formats_did_not_change' panicked at 'assertion failed: `(left == right)`
  left: `Some(Enum({0: Named { name: "UserTransaction", value: NewType(TypeName("SignedTransaction")) }, 1: Named { name: "WriteSet", value: NewType(TypeName("ChangeSet")) }, 2: Named { name: "BlockMetadata", value: NewType(TypeName("BlockMetadata")) }}))`,
 right: `Some(Enum({0: Named { name: "UserTransaction", value: NewType(TypeName("SignedTransaction")) }, 1: Named { name: "AuthenticatedWriteSet", value: NewType(TypeName("ChangeSet")) }, 2: Named { name: "BlockMetadata", value: NewType(TypeName("BlockMetadata")) }}))`: 
----
The recorded format for type `Transaction` is missing or does not match the recorded value on disk.
You may run `cargo run -p generate-format -- --record` to refresh the records.
Please verify the changes to the recorded file(s) and tag your pull-request as `breaking`.
----
```

**Comment by runtian-zhou** [comment]
As suggested by @sblackshear, `WaypointWriteSet` seems to be a clearer name to reflect that this is a WriteSet that gets applied with Waypoint and thus will never go through consensus

File: config/config-builder/src/validator_config.rs [hunk 4]
```diff
@@ -194,19 +194,16 @@ impl ValidatorConfig {
...
-                &validator_swarm.nodes,
-                validator_set,
-                discovery_set,
-                self.template
-                    .test
-                    .as_ref()
-                    .and_then(|config| config.publishing_option.clone()),
-            )
-            .into_inner(),
+        let genesis = Some(vm_genesis::encode_genesis_transaction_with_validator(
```
Comment (davidiw) [inline config/config-builder/src/validator_config.rs:197]: I think at some point, we're going to need to fashion a genesis builder much like we have for config builder, but I have a plan for that lol...

**Pull request closed without merging**